src/components/dialogs/dialogAlert/dialogAlertRoot/dialogAlertRoot.test.tsx

```diff
@@ -0,0 +1,115 @@
+import { render, screen } from '@testing-library/react';
+import { userEvent } from '@testing-library/user-event';
+import { DialogAlertContent, type IDialogAlertContentProps } from '../dialogAlertContent';
+import { DialogAlertFooter, type IDialogAlertFooterProps } from '../dialogAlertFooter';
+import { DialogAlertHeader, type IDialogAlertHeaderProps } from '../dialogAlertHeader';
+import { DialogAlertRoot, type IDialogAlertRootProps } from './dialogAlertRoot';
+
+describe('<DialogAlert.Root/> component', () => {
+    const createTestComponent = (
+        rootProps?: Partial<IDialogAlertRootProps>,
+        headerProps?: Partial<IDialogAlertHeaderProps>,
+        contentProps?: Partial<IDialogAlertContentProps>,
+        footerProps?: Partial<IDialogAlertFooterProps>,
```

Comment:
I wouldn't render the alert header, content & footer by default but just the eventual children property. In that way, we keep these unit tests independent from the implementation/properties of the alert header, content & footer components.
The `renders the alertdialog with the given content` test can simply check if the given children component get rendered when the dialog is open and the rests regarding the `onOpenChange` callback can be implemented by passing a `<Button />` component as children which triggers the `handleCancelBtnClick` callback on-click